pandas/core/internals/managers.py

```diff
@@ -1817,10 +1817,13 @@ def _shape_compat(x):
 
     first = arrays[0]
     shape = (len(arrays),) + _shape_compat(first)
-
-    stacked = np.empty(shape, dtype=dtype)
-    for i, arr in enumerate(arrays):
-        stacked[i] = _asarray_compat(arr)
+    if len(arrays) == 1:
+        # allow for 0-copy construction from dict
+        stacked = _asarray_compat(first).reshape(shape)
+    else:
+        stacked = np.empty(shape, dtype=dtype)
+        for i, arr in enumerate(arrays):
```

Comment:
why can't you use the original loop i think that will also allow 0-copy construction , no?

Reply:
Sorry, I don't follow what you mean by original loop.

The original loop I see is allocating memory and assigning into it, which doesn't allow 0-copy.